utils/firestore.js

```diff
@@ -431,6 +431,64 @@ const retrieveParticipantsEligibleForIncentives = async (siteCode, roundType, is
     }
 }
 
+const retrieveParticipantsDataDestruction = async (totalDocNumber = 10) => {
+    try {
+        const deletedFieldList = ['100767870', '113579866', '117249500', '121430614', '123868967', '126331570', '130371375', '131458944', '141450621', '142654897', '150818546', '153211406', '167958071', '175732191', '230663853', '253883960', '265193023', '271757434', '285488731', '311580100', '335767902', '352891568', '371067537', '372303208', '388711124', '404289911', '421823980', '430551721', '431428747', '452166062', '454205108', '454445267', '459098666', '471593703', '507120821', '512820379', '521824358', '523768810', '536735468', '538619788', '547363263', '564964481', '576083042', '596510649', '620696506', '624030581', '634434746', '637147033', '639172801', '646873644', '663265240', '684635302', '685002411', '693626233', '699625233', 'state'];
+        let pageLimit = 500;
+        let pageCount = 0;
+        let toContinue = true;
+        let docCount = 0;
+
+        pageLimit = Math.min(pageLimit, totalDocNumber);
+        const { isIsoDate } = require('./validation');
+        while (toContinue) {
+            const batch = db.batch();
+            const checkedDocCount = pageCount * pageLimit;
+            const remainingDocCount = totalDocNumber - checkedDocCount;
+            const currLimit = Math.min(remainingDocCount, pageLimit);
+
+            const currSnapshot = await db
+                .collection('participants')
+                .where('831041022', '==', 353358909)
+                .limit(currLimit)
+                .offset(checkedDocCount)
+                .get();
+
+            for (const doc of currSnapshot.docs) {
+                const participant = doc.data();
+                const millisecondsInDay = 24 * 60 * 60 * 1000;
+                const timeDiiff = isIsoDate(participant['269050420'])
+                    ? new Date().getTime() - new Date(participant['269050420']).getTime()
+                    : 0
+
+                if (participant['883668444'] === 704529432 || (Math.floor(timeDiiff / millisecondsInDay) > 60)) {
```

Comment:
I haven't read the DD SOP in awhile, is this the correct logic? The part that jumps out is "OR it's been more than 60 days since they requested a Data Destruction"

Reply:
@anthonypetersen the logic based on this description 
<img width="833" alt="image" src="https://github.com/episphere/connectFaas/assets/24559107/578b3483-26ab-4af5-9f67-274b1163b415">